scipy/stats/_stats.pyx

```diff
@@ -0,0 +1,31 @@
+#cython: boundscheck=False
+#cython: nonecheck=False
+#cython: wraparound=False
+#cython: cdivision=True
+from libc cimport math
+
+
+ctypedef fused real:
+    short
+    int
+    long
+    float
+    double
```

Comment:
It's just `int{8,16,32,64}_t`, `uint{8,16,32,64}_t`, `float{32,64}_t`, `longdouble_t`, `complex{64,128}_t`, `clongdouble_t`. I'm not sure if the object and string dtypes are supported in cython.

However, as noted, trying to make all of these work via fused types is 14*14=196 combinations --- so probably best to limit the dtypes to the common ones.